astroquery/casda/core.py

```diff
@@ -46,18 +45,69 @@ class CasdaClass(BaseQuery):
     URL = conf.server
     TIMEOUT = conf.timeout
     POLL_INTERVAL = conf.poll_interval
+    USERNAME = conf.username
     _soda_base_url = conf.soda_base_url
+    _login_url = conf.login_url
     _uws_ns = {'uws': 'http://www.ivoa.net/xml/UWS/v1.0'}
 
     def __init__(self, user=None, password=None):
         super(CasdaClass, self).__init__()
-        if user is None:
-            self._authenticated = False
+
+    def _login(self, *, username=None, store_password=False,
+               reenter_password=False, password=None):
```

Comment:
I'm a bit torn here as we're not at all consistent in the library as a whole, but having the password in the signature seems a bit of a counter-effective of security. I think I best like how the alma module is set up in this regard. What do you all think, am I just overly cautious? 

@keflavich  @andamian @pllim

Reply:
I added the password param to allow testing of the login as there is no keyring backend in the CI environment. I couldn't see anyone else testing the login path. I agree the password param should not be used by other users, so perhaps either a "For testing use only" note in the method's docstring, or hiding it in a **kwargs would be better?

Reply:
👎 to explicit password argument in API. Might as well set up the backend to handle anonymous login without a password, or enable a more secure way (encrypted token) for the CI. Or just not test this at all in the CI and set up something private server-side to test this securely.

Reply:
Backend changes are out of scope for this piece of work unfortunately. I've removed the password parameter and the tests for the login code. For the other test code which requires login I have added a utility to simulate the login result.

Reply:
Sounds good to me. Thanks!

Reply:
Thank you. I opened https://github.com/astropy/astroquery/issues/2391 with the aim of tracking this password in API issue for the other affected modules, and finding an acceptable backend solution for testing (very much beyond the scope for this PR) has already have one opened one in #2367.

Reply:
A consequence of removing the tests relying on the password parameter is that I can't achieve the code coverage target, so the CodeCov checks will fail unfortunately. The untested code occurs after the call to the keyring which would fail as there is no keyring backend in the GitHub CI environment.

Reply:
yeah, no worries about the coverage test, those are not run with remote-data anyway, so always show a much smaller percentage as the reality.

I'll open a follow-up issue to get a backend installed in the VM, it shouldn't be too hard.